added requested docstrings

AoifeHughes opened the pull request:
WRT: https://github.com/TuringLang/Turing.jl/issues/2684

Commit:
added requested docstrings

mhauru commented on src/mcmc/hmc.jl:
```diff
@@ -175,6 +214,41 @@ function find_initial_params(
+
+# Keyword Arguments
+
+- `initial_params`: Initial parameter values to use for sampling. If `nothing` (the default),
```
`initial_params` and `verbose` work the same way for all samplers, not just HMC, so I would not document them here, but rather refer to the docstring of the more generic `initialstep` function or method. That docstring may need writing too, but would make sense to postpone that after github.com/TuringLang/Turing.jl/pull/2676, because the way `initial_params` works is going to change quite drastically.

Commit:
Apply Mhauru's change suggestions to HMC documentation

- Improved docstring for initial_params to reference DynamicPPL.initialstep
- Added link to sampling options documentation for keyword arguments
- Fixed typos: "adaption" → "adaptation" throughout HMCDA and NUTS
